### ISSUE TITLE
A simple diode component

### ISSUE THREAD

**Issue by richardpawson** [CircuitVerse/CircuitVerse#1992]
I would like there to be a simple Diode component. This has one input and one output. With this truth table:

Input   Output
  0             Z (high impedance state)
  1             1

The current Tir-state component (very useful) is really like a Zener Diode (a diode switched by a third input)., I'm seeking a plain Diode.

The plain Diode is extremely important in order to be able to create a [diode matrix](https://en.wikipedia.org/wiki/Diode_matrix), where you have a grid of overlapping horizontal and vertical parallel wires, and where at each potential intersection you either join the H & V with a diode, or not. This is how simple encoders/decoders are designed in practice. The same thing could be done with a ROM (indeed a ROM is, effectively a large diode array) but an explicit diode array is much clearer.

**Comment by mayank-kabra2001** [comment]
Sir , in this if we design a diode , it is similar to buffer with no enable pin . As high impedance means we should not drive the signal in Analog but in digital it will simply means zero right ???

**Comment by richardpawson** [comment]
My (new) [Diode and Diode Matrix](https://circuitverse.org/users/56463/projects/diode-and-diode-matrix) CV project makes this clearer. It contains a simulation of a diode - built from a Tri-state where the input is also connected to the Enable pin - and an example of how multiple diodes are connected together in a diode matrix.

My simulated diode does the job. The advantage of making it a native CV component would be:

1) It would have the proper recognised symbol for a diode, instead of looking like a small sub-circuit.
2) It could be rotated, which a sub-circuit cannot.

That's it really.

**Comment by mayank-kabra2001** [comment]
I have done it .. will soon make the PR .

**Comment by richardpawson** [comment]
Great! Is it possible  that - before it gets to PR - that you could show a screen snippet showing what the diode looks like in action. (If that can't be done on GitHub, you could just post it on the Slack forum - I'll see it here.)

**Comment by mayank-kabra2001** [comment]
Yes sir , i am sharing screenshot of the following . If you want i can make video and upload it on slack . 

![Screenshot from 2020-12-09 15-06-37](https://user-images.githubusercontent.com/57330493/101612033-365f7980-3a30-11eb-8305-d328aac71f7d.png)
![Screenshot from 2020-12-09 15-06-41](https://user-images.githubusercontent.com/57330493/101612038-38293d00-3a30-11eb-8537-86b411f479ce.png)

**Comment by richardpawson** [comment]
Terrific - thanks! Small request. Could you make the visual representation of the diode even slightly smaller. Currently, it covers 3 x 3 grid squares. I think it could be 2 x 2. That would then make triangle part of the diode the same size as the triangle part of a NOT gate in CircuitVerse, which would be consistent and suitable as both are 1-input devices. (NOT gate is 2 x 3 squares, but it needs the circle on the end).  The main reason I am keen to see it made smaller is that a common use of diodes (in digital circuits) is to create a diode matrix (per my previous point), and you really then want the individual elements to be nice and compact.

**Comment by richardpawson** [comment]
Also, if you made it 2 x 2 it would be good if the triangle were solid black, as that is normal diode symbol. (However, if the CircuitVerse drawing routines do not facilitate solid shapes then this is not a big issue for me).

**Comment by mayank-kabra2001** [comment]
Done sir , made solid black and grid size covering is reduced .

**Comment by richardpawson** [comment]
Thanks!

**Comment by Palaksharma23** [comment]
i am working on this :)
